### PR TITLE
[go-server] Add dockerfile for building and run petstore service

### PR DESCRIPTION
**Pull request by etherealjoy**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [X] Filed the PR against the correct branch: Default: `master`.
- [X] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR
This PR aims to introduce a docker file to build and host the go server generated code inside a container.
Requests can be directed to container using API calls from a client.

- [X] Tested using curl on Linux

@antihax @wing328

### CODE REVIEW

**Comment by wing328** [comment]
cc @bvwells as well

File: modules/openapi-generator/src/main/resources/go-server/README.mustache [hunk 1]
```diff
@@ -28,3 +28,14 @@ To run the server, follow these simple steps:
...
 ```
 
+To run the server in a docker container
+```
+docker build --network=host -t petstore .
+```
+
+Once image is built use
+```
+docker run --rm -it petstore
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/go-server/README.mustache]: @etherealjoy should `petstore` be replaced by {{{packageName}}} instead?

Comment (etherealjoy) [reply modules/openapi-generator/src/main/resources/go-server/README.mustache]: indeed, I have updated the commit.

**Comment by wing328** [comment]
Tested locally and the result is good: (partial result)
```
Removing intermediate container 1cde8effa6b7
 ---> 3850dc3daa87
Step 11/11 : ENTRYPOINT ["./petstoreserver"]
 ---> Running in 04dc9f79b86d
Removing intermediate container 04dc9f79b86d
 ---> 0cca2b780ae2
Successfully built 0cca2b780ae2
Successfully tagged petstoreserver:latest
➜  go-api-server git:(etherealjoy-go-server_add_dockerfile) ✗ docker run --rm -it petstoreserver 
2018/06/13 08:53:57 Server started
```

**Pull request merged**